### PR TITLE
new numpy features for peak selection

### PR DESCRIPTION
**Pull request by h-mayorquin**
Updated the select peaks function to use the new -and recommended - features of numpy for random generator and container operations.

### CODE REVIEW

File: spikeinterface/sortingcomponents/peak_selection.py [hunk 1]
```diff
@@ -60,9 +62,8 @@ def select_peaks(peaks, method='uniform', seed=None, **method_kwargs):
     """
 
     selected_peaks = []
-    
-    if seed is not None:
-        np.random.seed(seed)
+    seed = seed if seed else None
+    rng = np.random.default_rng(seed=seed)
```
Comment (samuelgarcia) [inline spikeinterface/sortingcomponents/peak_selection.py:66]: Cool!

**Comment by samuelgarcia** [comment]
This OK for me.
You do not like `in1d` ?

**Comment by h-mayorquin** [comment]
> This OK for me. You do not like `in1d` ?

@samuelgarcia 
I think `isin` reflects intent better. Plus it is also is used in pandas and alludes to the `in` operator in python. Finally, this is the suggestion in the numpy documentation:
https://numpy.org/doc/stable/reference/generated/numpy.in1d.html

**Comment by h-mayorquin** [comment]
@alejoe91 I solved the merge conflicts with #1318

**Comment by alejoe91** [comment]
Awesome! We should also use the rng approach in other parts of the code..I think that in several places we still use the old method

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**